### DISCUSSION TITLE
[Question]  PDF - How to apply style/class to related list

### DISCUSSION THREAD

**Discussion by TonyM15** [YetiForceCompany/YetiForceCRM#11493] [How to]
I need to wrap each field (of related list)  in box (border). Can I use styles and how?

**Comment by mad99** [comment]
Have done a search in source-code and found the ***/vendor/yetiforce/yetiforcepdf/lib/Style/Style.php***

By trial&error this file seams to be the css-stylesheet root for PDF-style.
### CHANGE:
![PDF-Stylsheet](https://user-images.githubusercontent.com/6169308/65120610-30761680-d9ee-11e9-8be6-5e834a4b2bbe.png)
### RESULT:
![Bildschirmfoto 2019-09-18 um 08 51 30](https://user-images.githubusercontent.com/6169308/65122558-9021f100-d9f1-11e9-9a92-b38f1e56dd6d.png)


## BUT MORE INTERESTING: How can we use this?
So this file seams to be the base for css-stylsheet for PDF.
But what we can do with?
- ***Is it thought to change the Style.php file itself?***
- **Is there some inheritance to other files, we could use to create the own  base?**
- **How it is possible or the best way to inherit from this file?** Because of the «protected» property I guess the «own stylsheet-file has to extends the «Style» Class.
- ***Is there an example somewhere for do this in the right approach?***
- **Is there some API-stuff like a command to insert a particular own css to a particular PDF-Template?** Would be great to have different styles for different invoices/correcting invoices/quotes/...

Would be great to get some insights or help/hints from more experienced users.
Thanks 
Martin

**Comment by waran70** [comment]
Works with invoices but not with related records lists

**Comment by waran70** [comment]
Martin, did you managed to find solution for independent PDF styling?

**Comment by mad99** [comment]
Hey waran70,
At the end it was easier for me to do it from scratch.
So I wrote a MasterClass with my own functions and hardcoded html-printout with CSS in php-code. Therein I wrote own sql-queries copied from other classes found in YetiForm and own limited knowledge. 

So the invoice-modul is the main-reason why I have mixed feeling with YetiForce.
But it's free. Soi it's just brilliant. But the amount of work is also correspondingly high.

> If you like, I can send you the code in a PM. It works for all my invoices and qoutes.
But all in all the solution is very BETA. Also unfortunately to change some part of it's a hard lift most of the time. The abstraction layer is not deep enough thought. A lot of shame-code has been added through all the extra workarounds.

> In my code there is no peace like it was in the Yeti-Form Examples. Also my YetiForce CRM is not last version. So you would be leaved at your own to fix some bug's for your YF.
I can't guarantee for any of the damages, because of the hard-coded SQL-Queries. 
So it would only be manageable if you have a good base of SQL-query and PHP code knowledge.
You have to scan the code before you try. Or you have to try it out in a nutshell seperated from your real YF or data.
That's why I wouldn't publish it here.


So, yes I have found «my» solution for independing PDF style. 
And it works for my like a charme - for me. 
But it's a hard lift, if some part has to be different, because of the most output is hardcoded and a lot of stuff is Beta(Maybe buggy in other YF-Environments ...  
So, I would never recommend it to anyone as a general solution!
